qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -947,6 +948,85 @@ def _gt_k() -> GreaterThanConstant:
 _GREATER_THAN_K_DOC = BloqDocSpec(bloq_cls=GreaterThanConstant, examples=[_gt_k])
 
 
+@frozen
+class Equals(Bloq):
+    r"""Implements |x>|y>|t> => |x>|y>|t ⨁ (x = y)> using $n-1$ Toffoli gates.
+
+    Args:
+        dtype: Data type of the input registers `x` and `y`.
+
+    Registers:
+        x: First input register.
+        y: Second input register.
+        target: Register to hold result of comparison.
+    """
+
+    dtype: QDType
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build_from_dtypes(x=self.dtype, y=self.dtype, target=QBit())
+
+    @cached_property
+    def bitsize(self) -> SymbolicInt:
+        return self.dtype.num_qubits
+
+    def is_symbolic(self):
+        return is_symbolic(self.dtype)
+
+    def wire_symbol(self, reg: Optional[Register], idx: Tuple[int, ...] = tuple()) -> WireSymbol:
+        if reg is None:
+            return Text("")
+        if reg.name == 'x':
+            return TextBox("In(x)")
+        if reg.name == 'y':
+            return TextBox("In(y)")
+        elif reg.name == 'target':
+            return TextBox("⨁(x = y)")
+        raise ValueError(f'Unknown register symbol {reg.name}')
+
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', x: 'Soquet', y: 'Soquet', target: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
+        if self.is_symbolic():
```

Comment:
either explicitly test that self.bitsize is symblic here or add the type guard https://github.com/quantumlib/Qualtran/blob/db8b9938d8210024bb6b92942376363586165efc/qualtran/symbolics/types.py#L73